integration-tests/pkg/mock_sensor/server.go

```diff
@@ -171,8 +166,11 @@ func (m *MockSensor) HasConnection(containerID string, conn types.NetworkInfo) b
 	defer m.networkMutex.Unlock()
 
 	if conns, ok := m.connections[containerID]; ok {
-		_, exists := conns[conn]
-		return exists
+		for _, connection := range conns {
+			if connection.Equal(conn) {
+				return true
+			}
+		}
```

Comment:
I think we can use [slices.ContainsFunc](https://pkg.go.dev/slices#ContainsFunc) here:
```suggestion
        return slices.ContainsFunc(conns, func(c Connection) bool {
            return c.Equal(conn)
        })
```

Reply:
Done. Though I didn't commit this exact change, because there is an error in it.